src/libraries/Microsoft.Extensions.Options/src/UnnamedOptionsManager.cs

```diff
@@ -1,22 +1,35 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Threading;
 
 namespace Microsoft.Extensions.Options
 {
-    internal class UnnamedOptionsManager<[DynamicallyAccessedMembers(Options.DynamicallyAccessedMembers)] TOptions> :
+    internal sealed class UnnamedOptionsManager<[DynamicallyAccessedMembers(Options.DynamicallyAccessedMembers)] TOptions> :
         IOptions<TOptions>
         where TOptions : class
     {
-        private readonly Lazy<TOptions> _lazy;
+        private readonly IOptionsFactory<TOptions> _factory;
+        private volatile object _syncObj;
+        private volatile TOptions _value;
 
-        public UnnamedOptionsManager(IOptionsFactory<TOptions> factory)
+        public UnnamedOptionsManager(IOptionsFactory<TOptions> factory) => _factory = factory;
+
+        public TOptions Value
         {
-            _lazy = new Lazy<TOptions>(() => factory.Create(Options.DefaultName));
-        }
+            get
+            {
+                if (_value is TOptions value)
```

Comment:
For my education, is there a significant difference here vs `if (_value is not null) return _value;`?

Reply:
I guess because the field is marked `volatile` it's better to read it once?

Reply:
The main difference is whether the field is read twice rather than once.  With it being volatile, the JIT can't elide the second read, and on ARM, it would be another barrier.